libeos-updater-util/config.c

```diff
@@ -108,25 +125,69 @@ euu_config_file_set_property (GObject      *object,
     {
     case PROP_PATHS:
       /* Construct-only; must be non-empty */
+      g_assert (self->paths == NULL);
       self->paths = g_value_dup_boxed (value);
       g_assert (self->paths != NULL);
       self->n_paths = g_strv_length (self->paths);
       g_assert (self->n_paths > 0);
       break;
 
+    case PROP_DEFAULT_RESOURCE:
+      /* Construct-only. Must be non-%NULL. */
+      g_assert (self->default_resource == NULL);
+      self->default_resource = g_value_dup_boxed (value);
+      g_assert (self->default_resource != NULL);
+      break;
+
+    case PROP_DEFAULT_PATH:
+      /* Construct only; must be non-%NULL. */
+      g_assert (self->default_path == NULL);
+      self->default_path = g_value_dup_string (value);
+      g_assert (self->default_path != NULL);
+      break;
+
     default:
       G_OBJECT_WARN_INVALID_PROPERTY_ID (object, property_id, spec);
       break;
     }
 }
 
 static void
+euu_config_file_constructed (GObject *object)
+{
+  EuuConfigFile *self = EUU_CONFIG_FILE (object);
+  g_autoptr(GBytes) bytes = NULL;
+  g_autoptr(GError) error = NULL;
+
+  /* Chain up. */
+  G_OBJECT_CLASS (euu_config_file_parent_class)->constructed (object);
+
+  /* Load the default config file from the given resource. It’s a fatal error
+   * if this fails. We load this in the constructor to ensure we fail early,
+   * rather than conditionally on accessing something from the config file. */
```

Comment:
You could also make this implement `GInitable` and enforce it that way -- but this is OK too.